### PR TITLE
Improve contrast for icons in `invisible` button on hover in dark mode

### PR DESCRIPTION
**Pull request by langermank**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?

- Update `Button` CSS to use new tokens for icon colors in the `invisible` variant
- Sneaking in a fix for v8 colors to remove the unused inset box-shadow on `primary` buttons

### Screenshots

| Before | After |
| -- | -- |
| icon button hover | icon button hover |

### Integration

- Bump Primitives to the latest version

#### List the issues that this change affects.

Closes https://github.com/github/accessibility-audits/issues/5764

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **Fixes axe scan violation** - This change fixes an existing [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation.

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Lookbook)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Review by keithamus — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
     "@github/prettier-config": "0.0.4",
     "@playwright/test": "^1.35.1",
     "@primer/css": "21.0.2",
-    "@primer/primitives": "7.12.0",
+    "@primer/primitives": "0.0.0-20230921000240",
```
Comment (jonrohan) [inline package.json:65]: Shouldn't we fix this to a release

Comment (langermank) [reply package.json:65]: Yes, I need to make another patch release before I merge this PR

File: app/components/primer/beta/button.pcss [hunk 2]
```diff
@@ -247,8 +247,16 @@ summary.Button {
 .Button--invisible {
   color: var(--button-default-fgColor-rest);
 
+  &.Button--iconOnly {
+    color: var(--button-invisible-iconColor-rest);
```
Comment (langermank) [inline app/components/primer/beta/button.pcss:251]:
```suggestion
    color: var(--button-invisible-iconColor-rest, var(--color-fg-muted);
```

File: app/components/primer/beta/button.pcss [hunk 3]
```diff
@@ -247,8 +247,16 @@ summary.Button {
...
+  }
+
   &:hover:not(:disabled) {
     background-color: var(--button-invisible-bgColor-hover);
+
+    & .Button-visual {
+      color: var(--button-invisible-iconColor-hover);
```
Comment (langermank) [inline app/components/primer/beta/button.pcss:258]:
```suggestion
      color: var(--button-invisible-iconColor-hover, var(--color-fg-default));
```

File: app/components/primer/beta/button.pcss [hunk 4]
```diff
@@ -270,7 +278,7 @@ summary.Button {
   }
 
   & .Button-visual {
-    color: var(--fgColor-muted);
+    color: var(--button-invisible-iconColor-rest);
```
Comment (langermank) [inline app/components/primer/beta/button.pcss:281]:
```suggestion
    color: var(--button-invisible-iconColor-rest, var(--color-fg-muted));
```

**Review by jonrohan — APPROVED** [review]

**Pull request merged**